### PR TITLE
Port to VisualStudio compiler and Win-10 (not complete)

### PR DESCRIPTION
**Pull request by mrkkrj**
- added support for Microsoft compiler 
- fixed crash on Win-64
- added support for time measurements to run.py on Win-10

OPEN TODO: run.py build support for Windows!!!
 --> But if you build manually, you can use the run.py script to compare results

### CODE REVIEW

**Comment by mrkkrj** [comment]
New commit added - building on Windows possible with the run.py script.
Problem -> validation fails bc of different file sizes!

File: run.py [hunk 1]
```diff
@@ -74,15 +74,29 @@
...
+          print("  copy .exe - Failed")
 
       inputFileName = str(os.path.join(saveCwd, "221575.pgm"))
       destDir = str(submissionBuildDir)
       try:
-        subprocess.check_call("cp " + inputFileName + " " + destDir, shell=True, stdout=FNULL, stderr=FNULL)
+        if sys.platform != 'win32':                
+          subprocess.check_call("cp " + inputFileName + " " + destDir, shell=True, stdout=FNULL, stderr=FNULL)
+        else:
+            subprocess.check_call("copy " + inputFileName + " " + destDir, shell=True, stdout=FNULL, stderr=FNULL)
```
Comment (DenisBakhvalov) [inline run.py]:
```suggestion
          subprocess.check_call("copy " + inputFileName + " " + destDir, shell=True, stdout=FNULL, stderr=FNULL)
```

File: run.py [hunk 2]
```diff
@@ -108,39 +122,58 @@
...
-
+        if sys.platform != 'win32':
+          output = subprocess.check_output("time -p " + runCmd, shell=True)
+          for row in output.split(b'\n'):
+            if b'real' in row:
+              real, time = row.split(b' ')
+              scores.append(float(time))          
+        else:
+          print("  run - WIN 32")
+          output = subprocess.check_output("powershell -Command \"Measure-Command {" + runCmd + " | Out-Default}\" | findstr TotalSeconds" , shell=True)
```
Comment (DenisBakhvalov) [inline run.py:171]: Do you know if powershell is available by default?

**Review by DenisBakhvalov — COMMENTED** [review]
Thanks, Marek! The changes look good.

> Problem -> validation fails bc of different file sizes!
That's interesting. Will you find the time to take a look at it, please?

**Comment by mrkkrj** [comment]
I'll have a look today...
Mrk.

**Comment by mrkkrj** [comment]
The problem was that fprintf uses CLRF on Windows!

File: run.py (hunk 2 above)
Comment (mrkkrj) [reply run.py:171]: Docs says so:

"Windows PowerShell comes installed by default in every Windows, starting with Windows 7 SP1 and Windows Server 2008 R2 SP1."

https://docs.microsoft.com/de-de/powershell/scripting/windows-powershell/install/installing-windows-powershell?view=powershell-7.1

Comment (DenisBakhvalov) [reply run.py:171]: Ok, thanks!

File: canny_baseline/canny_source.c [hunk 3]
```diff
@@ -975,7 +975,11 @@ int write_pgm_image(char *outfilename, unsigned char *image, int rows,
    ***************************************************************************/
    if(outfilename == NULL) fp = stdout;
    else{
+#if _WIN32
```
Comment (DenisBakhvalov) [inline canny_baseline/canny_source.c:978]: @ibogosavljevic, please review Marek's changes in canny_source.c
Do they conflict with your changes in #5?

**Comment by ibogosavljevic** [comment]
They dont conflict, I modified only validate.cpp.

All Marek's changes are inside #ifdef Win32, he didn't touch the common
code so I guess they are safe to land.

**Comment by DenisBakhvalov** [comment]
Thanks, @ibogosavljevic .
@mrkkrj do you plan more changes in this PR?

**Comment by mrkkrj** [comment]
No, at the moment I don't have time to play with it more. But the basic usage case is running - at least for me with Visual Studio 2019.

**Pull request merged**